File: app/src/main/java/net/luanwilliam/cookbook/AbrirCategorias.java
package net.luanwilliam.cookbook;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import net.luanwilliam.cookbook.DAO.ReceitaDatabase;
import net.luanwilliam.cookbook.model.Receita;

import java.util.ArrayList;
import java.util.List;

public class AbrirCategorias extends AppCompatActivity {


    private  ListView listviewReceitas;
    private TextView txtNome, txtDesc;
    private long idCat, idRec;
    private List<Receita> listaReceitas;
    private ArrayAdapter<Receita> adapterReceitas;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_abrir_categorias);
        Intent intent = getIntent();

        Bundle param = intent.getExtras();
        long ide = param.getLong("id");
        setIdCat(ide);

        listviewReceitas = findViewById(R.id.listaReceitas);

        String nome = param.getString("nome");
        txtNome = findViewById(R.id.title);

        txtNome.setText("Receitas cadastradas na categoria: "+nome);

        String desc = param.getString("descricao");
        carregaReceitas(getIdCat());




     listviewReceitas.setOnItemClickListener(((parent, view, position, id) -> {

            Receita rec = (Receita) parent.getItemAtPosition(position);


         System.out.println("Teste id "+rec.getIdrec());

          /*  System.out.println("TESTE POSITION NOME "+rec.getNome());
            System.out.println("rec descricao "+rec.getDescricao());
            System.out.println("ingredientes "+rec.getListaIngredientes());
            System.out.println("modo de preparoo "+rec.getModoPreparo()); */


            Bundle params = new Bundle();
            params.putLong("id", rec.getIdrec());
            params.putString("nome", rec.getNome());
            params.putString("descricao", rec.getDescricao());
            params.putString("ingredientes", rec.getListaIngredientes());
            params.putString("modo_preparo", rec.getModoPreparo());
            params.putString("data", rec.getData());

            Intent intencao = new Intent(AbrirCategorias.this, AbrirReceitas.class);

            intencao.putExtras(params);

            startActivity(intencao);
            finish();


        }));


    }

    public long getIdCat() {
        return idCat;
    }

    public void setIdCat(long idCat) {
        this.idCat = idCat;
    }

    public long getIdRec() {
        return idRec;
    }

    public void setIdRec(long idRec) {
        this.idRec = idRec;
    }

    public void carregaReceitas(long idCat){

        AsyncTask.execute( () ->{
            ReceitaDatabase db = ReceitaDatabase.getDatabase(AbrirCategorias.this);
            listaReceitas = db.receitaDAO().listarReceitasIdCategoria(idCat);




            AbrirCategorias.this.runOnUiThread( () ->{
                adapterReceitas = new ArrayAdapter<>(AbrirCategorias.this,
                        android.R.layout.simple_list_item_1,listaReceitas );

                listviewReceitas.setAdapter(adapterReceitas);

            });




        });




    }


}

File: app/src/main/java/net/luanwilliam/cookbook/AbrirCategorias.kt
package net.luanwilliam.cookbook

import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import net.luanwilliam.cookbook.dao.ReceitaDatabase
import net.luanwilliam.cookbook.model.Receita

class AbrirCategorias : AppCompatActivity() {
    private var listviewReceitas: ListView? = null
    private var txtNome: TextView? = null
    private val txtDesc: TextView? = null
    var idCat: Long = 0
    var idRec: Long = 0
    private var listaReceitas: List<Receita>? = null
    private var adapterReceitas: ArrayAdapter<Receita>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_abrir_categorias)
        val intent = intent
        val param = intent.extras
        val ide = param!!.getLong("id")
        idCat = ide
        listviewReceitas = findViewById(R.id.listaReceitas)
        val nome = param.getString("nome")
        txtNome = findViewById(R.id.title)
        txtNome!!.text = "Receitas cadastradas na categoria: $nome"
        carregaReceitas(idCat)
        listviewReceitas!!.setOnItemClickListener { parent: AdapterView<*>, view: View?, position: Int, id: Long ->
            val rec = parent.getItemAtPosition(position) as Receita
            println("Teste id " + rec.idrec)
            val params = Bundle()
            params.putLong("id", rec.idrec)
            params.putString("nome", rec.nome)
            params.putString("descricao", rec.descricao)
            params.putString("ingredientes", rec.listaIngredientes)
            params.putString("modo_preparo", rec.modoPreparo)
            params.putString("data", rec.data)
            val intencao = Intent(this@AbrirCategorias, AbrirReceitas::class.java)
            intencao.putExtras(params)
            startActivity(intencao)
            finish()
        }
    }

    fun carregaReceitas(idCat: Long) {
        AsyncTask.execute {
            val db = ReceitaDatabase.getDatabase(this@AbrirCategorias)
            listaReceitas = db!!.receitaDAO()!!.listarReceitasIdCategoria(idCat) as List<Receita>?
            runOnUiThread {
                adapterReceitas = ArrayAdapter(
                    this@AbrirCategorias,
                    android.R.layout.simple_list_item_1, listaReceitas!!
                )
                listviewReceitas!!.adapter = adapterReceitas
            }
        }
    }
}